bakery/pipeline-by-jobs.yml

```diff
@@ -0,0 +1,221 @@
+          channel: "{{ce-bot/channel}}"
+          thread_ts: "{{ce-bot/ts}}"
+      - get: s3
+      - get: cnx-recipes
+      - task: assemble-book-via-neb
+        file: pdf-spike/bakery/tasks/t-assemble-book.yml
+        on_failure:
+          put: ce-bot-message
+          params:
+            text: "Failed to assemble book `{{book-title/name}}`\n\n```\n{{book-title/stderr}}```"
+            channel: "{{ce-bot/channel}}"
+            thread_ts: "{{ce-bot/ts}}"
+        on_success:
+          put: s3
+          params:
+            path: book
+
+  - name: bake-book
+    plan:
+      - get: ce-bot
+        passed: [assemble-book]
+        trigger: true
+      - get: pdf-spike
+      - task: look-up-book
+        file: pdf-spike/bakery/tasks/t-book-lookup.yml
+      - put: ce-bot-message
+        params:
+          text: "Baking book `{{book-title/name}}`"
+          channel: "{{ce-bot/channel}}"
+          thread_ts: "{{ce-bot/ts}}"
+      - get: s3
+      - get: cnx-recipes
+      - task: bake-book-via-easy-bake
+        file: pdf-spike/bakery/tasks/t-bake-book.yml
+        on_failure:
+          put: ce-bot-message
+          params:
+            text: "Failed to bake book `{{book-title/name}}`\n\n```\n{{book-title/stderr}}```"
+            channel: "{{ce-bot/channel}}"
+            thread_ts: "{{ce-bot/ts}}"
+        on_success:
+          put: s3
+          params:
+            path: book
+
+  - name: mathify-book
+    plan:
+      - get: ce-bot
+        passed: [bake-book]
+        trigger: true
+      - get: pdf-spike
+      - task: look-up-book
+        file: pdf-spike/bakery/tasks/t-book-lookup.yml
+      - put: ce-bot-message
+        params:
+          text: "Mathifying book `{{book-title/name}}`"
+          channel: "{{ce-bot/channel}}"
+          thread_ts: "{{ce-bot/ts}}"
+      - get: s3
+      - task: mathify-book-via-mathify
+        file: pdf-spike/bakery/tasks/t-mathify-book.yml
+        on_failure:
+          put: ce-bot-message
+          params:
+            text: "Failed to mathify book `{{book-title/name}}`\n\n```\n{{book-title/stderr}}```"
+            channel: "{{ce-bot/channel}}"
+            thread_ts: "{{ce-bot/ts}}"
+        on_success:
+          put: s3
+          params:
+            path: book
+
+  - name: generate-pdf
+    plan:
+      - get: ce-bot
+        passed: [mathify-book]
+        trigger: true
+      - get: pdf-spike
+      - task: look-up-book
+        file: pdf-spike/bakery/tasks/t-book-lookup.yml
+      - put: ce-bot-message
+        params:
+          text: "Building pdf for `{{book-title/name}}`"
+          channel: "{{ce-bot/channel}}"
+          thread_ts: "{{ce-bot/ts}}"
+      - get: s3
+      - task: pdf-book-via-princexml
+        file: pdf-spike/bakery/tasks/t-build-pdf.yml
+        on_failure:
+          put: ce-bot-message
+          params:
+            text: "Failed to build pdf for book `{{book-title/name}}`\n\n```\n{{s3/book/stderr}}```"
```

Comment:
Why is this one `s3/book/stderr` instead of `book-title/stderr` (to be changed to `book/stderr`) like all the other ones?